### PR TITLE
Add configuration for network error handling

### PR DESCRIPTION
**Pull request by RikudouSage**
#23 #24 

@shadowhand Any comments?

### CODE REVIEW

File: src/DynamoDbCache.php [hunk 1]
```diff
@@ -582,21 +593,29 @@ private function getKey(string $key): string
...
+                case NetworkErrorMode::IGNORE:
+                    throw new CacheItemNotFoundException('', 0, $e);
+                case NetworkErrorMode::THROW:
+                    throw $e;
+                case NetworkErrorMode::WARNING:
+                    trigger_error("Network error when connecting to DynamoDB: {$e->getMessage()}", E_USER_WARNING);
+                    break; // @codeCoverageIgnore
+            }
+
+            throw new LogicException("Unsupported network error mode: {$this->networkErrorMode}");
```
Comment (shadowhand) [inline src/DynamoDbCache.php:618]: Seems like this should be done in the constructor and/or the builder.

File: src/DynamoDbCache.php [hunk 2]
```diff
@@ -582,21 +593,29 @@ private function getKey(string $key): string
      */
     private function getRawItem(string $key): array
     {
-        $input = [
-            'Key' => [
-                $this->primaryField => [
-                    'S' => $key,
-                ],
-            ],
-            'TableName' => $this->tableName,
```
Comment (shadowhand) [inline src/DynamoDbCache.php:591]: I moved this out so that it would be clear it couldn't cause any exceptions.

**Review by shadowhand — APPROVED** [review]
Looks good. Couple of comments but this would be an acceptable compromise.

File: src/DynamoDbCache.php [hunk 3]
```diff
@@ -582,21 +593,29 @@ private function getKey(string $key): string
...
             return $item->getItem();
         } catch (NetworkException $e) {
-            throw new CacheItemNotFoundException('', 0, $e);
+            switch ($this->networkErrorMode) {
+                case NetworkErrorMode::IGNORE:
+                    throw new CacheItemNotFoundException('', 0, $e);
+                case NetworkErrorMode::THROW:
+                    throw $e;
+                case NetworkErrorMode::WARNING:
+                    trigger_error("Network error when connecting to DynamoDB: {$e->getMessage()}", E_USER_WARNING);
```
Comment (shadowhand) [inline src/DynamoDbCache.php:614]: Another option would be to depend on PSR-3 and log that way. 🤷🏼‍♂️

File: src/DynamoDbCache.php (hunk 2 above)
Comment (RikudouSage) [reply src/DynamoDbCache.php:591]: Makes sense.

File: src/DynamoDbCache.php (hunk 3 above)
Comment (RikudouSage) [reply src/DynamoDbCache.php:614]: May add that in the future.

**Comment by shadowhand** [comment]
@RikudouSage thank you!

**Pull request merged**